common/src/IO/AssimpParser.cpp

```diff
@@ -327,6 +343,126 @@ std::optional<size_t> getMeshIndex(const aiScene& scene, const aiMesh& mesh)
   return std::nullopt;
 }
 
+/**
+ * Gets the channel index for a particular assimp node, matched by name.
+ */
+std::optional<size_t> getChannelIndex(const aiAnimation& animation, const aiNode& node)
+{
+  for (unsigned int i = 0; i < animation.mNumChannels; ++i)
+  {
+    const aiNodeAnim* ch = animation.mChannels[i];
+    if (!std::strcmp(node.mName.data, ch->mNodeName.data))
+    {
+      return size_t(i);
+    }
+  }
+  return std::nullopt;
+}
+
+/**
+ * Computes the animation information for each channel in an
+ * animation sequence. Always uses the first frame of the animation.
+ */
+std::vector<AssimpBoneInformation> getAnimationInformation(
+  const aiNode& root, const aiAnimation& animation)
+{
+  auto indivTransforms = std::vector<aiMatrix4x4>{animation.mNumChannels};
+
+  // calculate the transformations for each animation channel
+  for (unsigned int i = 0; i < animation.mNumChannels; ++i)
+  {
+    const aiNodeAnim* channel = animation.mChannels[i];
+
+    aiVector3D position(0, 0, 0);
```

Comment:
Always use assignment style and brace initialization syntax: `auto position = aiVector3D{0, 0, 0};`

Reply:
In fact, I would prefer this:

```
const auto position = channel.mNumPositionKeys > 0 ? channel.mPositionKeys[0].mValue : aiVector{0, 0, 0};
```

Then you can make the value const.